src/utils/getResponse.ts

```diff
@@ -85,7 +85,9 @@ export const getResponse = async <
 
       const mockedResponse = await resolver(publicRequest, response, context)
 
-      if (!mockedResponse) {
+      // Skip a handler if there is no response or if it has been already used for
+      // a one-time response.
+      if (!mockedResponse || requestHandler.shouldSkip) {
```

Comment:
So because there can be multiple one-time responses running in parallel, we don't guarantee the request handler's relevancy in the preceding `.filter` step, right?

Reply:
Yea, correct. At the time when the `filter` is invoked the two requests are still pending, thus the `filter` doesn't know it has to be skipped.
At least that's how I understand it. 

It does bring the following question/problem tho, how do we decide which response handler handles the first request? The one that receives the request first, or the one that returns the response first. 
With this implementation it's the handler that returns first, which is fine for me.

Reply:
It would be nice to include this short motivation note above the logic. Things like these usually take some playing around to figure out.

Reply:
How about that? Too verbose? Or it's ok?

```ts
  const relevantHandlers = handlers
    .filter((requestHandler) => {
      // Skip a handler if it has been already used for a one-time response.
      // Because there can be multiple one-time responses running in parallel,
      // we can't guarantee the request handler's relevancy here.
      // Thus, an extra check in succeeding `relevantHandlers` reducer.
      // The filter is kept here as well, so the handler is skipped sooner
      // for sequential requests.
      return !requestHandler.shouldSkip
    })
```

```ts
      // Skip a handler if one of the following is true:
      // * There is no response
      // * It has been already used for a one-time response. Because there can be multiple
      //   one-time responses running in parallel, we can't guarantee the request handler's
      //   relevancy in the preceding `.filter` step. Thus an extra check here.
      if (!mockedResponse || requestHandler.shouldSkip) {
        return acc
      }
```

Reply:
I like that, but it's too verbose. We can leave the `filter` comment out and only comment on the part we've added. 

```js
// Check for the handler revelancy
// in case of multiple parallel one-time handlers.
```

It doesn't need to be an in-detail explanation, as we have Git history to help us. A quick comment is still useful, so the logic won't get mistaken by a repetition, and to stress that it's intentional.

Reply:
Cool, I've updated it, please have another look!